dotcom-rendering/src/components/Analytics.amp.tsx

```diff
@@ -29,6 +29,48 @@ export const Analytics = ({
 }: Props) => {
 	const scripts: string[] = [
 		`<amp-analytics config="https://ophan.theguardian.com/amp.json" data-credentials="include" ></amp-analytics>`,
+		`<amp-analytics config="https://ophan.theguardian.com/amp.json" data-block-on-consent="_till_responded" data-credentials="include">
+			<script type="application/json">
+				{
+					"requests": {
+						"pageViewWithConsentTest": "http://localhost:8001/receive?componentEvent=\${componentEvent}",
+						"pageViewWithConsent": "\${additionalBase}&componentEvent=\${componentEvent}"
+					},
+					"triggers": {
+						"trackPageviewTcf": {
+							"on": "visible",
+							"request": "pageViewWithConsentTest",
+							"selector": ".amp-geo-group-tcfv2",
+							"vars": {
+								"componentEvent": "%7B%22consentJurisdiction%22:%22TCF%22,%22consentUUID%22:%22\${clientId(consentUUID)}%22,%22consent%22:%22\${consentString}%22%7D"
```

Comment:
I think that this escaped string can be best written with `encodeURI` or `encodeURIComponent` to help with maintainability: 

```js
const componentEvent = encodeURI(JSON.stringify({
  consentJurisdiction: "TCF",
  consentUUID: "CLIENT_ID",
  consent: "CONSENT_STRING",
}))
  .replace("CLIENT_ID", "\\${clientId(consentUUID)}")
  .replace("CONSENT_STRING", "\\${consentString}")
```

Reply:
Thanks for the feedback. It's a lot clearer this way.